eras/shelley/impl/src/Cardano/Ledger/Shelley/API/Genesis.hs

```diff
@@ -35,40 +34,49 @@ import Data.Kind (Type)
 import qualified Data.Map.Strict as Map
 
 -- | Indicates that this era may be bootstrapped from 'ShelleyGenesis'.
-class CanStartFromGenesis era where
+class
+  ( EraTxOut era
+  , Default (State (EraRule "PPUP" era))
```

Comment:
Nah, this `Default` constraint will just change to whatever the `Default` of type of this field will be:

https://github.com/input-output-hk/cardano-ledger/blob/9bb89710943cf14cca968551a935f1c7c058cf1c/eras/shelley/impl/src/Cardano/Ledger/Shelley/LedgerState/Types.hs#L257

Reply:
> maybe it's a better idea to create a default signature for that function

Then you'll have to create a new implementation for that function for no reason for all subsequent eras starting with Conway. More over, this is not the only place where this problem will appear, you'll have to solve in all places where constraint for `State (EraRule "PPUP" era)` appears. Here are just a few examples:

https://github.com/input-output-hk/cardano-ledger/blob/9bb89710943cf14cca968551a935f1c7c058cf1c/eras/shelley/impl/src/Cardano/Ledger/Shelley/LedgerState/Types.hs#L120-L122

https://github.com/input-output-hk/cardano-ledger/blob/9bb89710943cf14cca968551a935f1c7c058cf1c/eras/shelley/impl/src/Cardano/Ledger/Shelley/LedgerState/Types.hs#L128-L130

https://github.com/input-output-hk/cardano-ledger/blob/9bb89710943cf14cca968551a935f1c7c058cf1c/eras/shelley/impl/src/Cardano/Ledger/Shelley/LedgerState/Types.hs#L148-L150

Reply:
In fact this constraint comes from here:

https://github.com/input-output-hk/cardano-ledger/blob/9bb89710943cf14cca968551a935f1c7c058cf1c/eras/shelley/impl/src/Cardano/Ledger/Shelley/LedgerState/Types.hs#L508-L509

So once it is change in the `Default` instance `UTxOState`, it will change here as well